### DISCUSSION TITLE
Navigation options are not working correctly

### DISCUSSION THREAD

**Discussion by avs-doom** [remix-run/react-router#11923] [Q&A]
All hi!

I I'm trying to understand why in version 6.26.1 the behavior of the settings in navigation does not affect at all, although in version 6.16.0 everything works fine for me. At the moment I can't update the package because of this problem. My problem was closed without giving me an exhaustive answer, maybe here they can help me) Thank you. An example of what does not work correctly:

https://codesandbox.io/p/devbox/cranky-sammet-spfyy3

You can see that the settings are different, but the address changes the same way. In version 6.16.0, the addresses change correctly in my opinion.

**Comment by brophdawg11** [comment]
All react router routing is relative to the [current route hierarchy](https://reactrouter.com/en/main/components/link#relative) where it is called.  You're calling `useNavigate` on the root component so you're unable to route upwards since you're already at the top - so `..` is useless where you are using it.

If you move the `navigate` calls down into a route lower in the hierarchy you can see the difference between the `relative` option behaviors: https://codesandbox.io/p/devbox/patient-cherry-gf2dty?file=%2Fsrc%2FApp.tsx%3A33%2C41.

`relative: 'route'` is the default and goes up one `<Route>` in the hierarchy - so `../test2` from `/test/123` lands on `/test2`

`relative: 'path'` is an opt in behavior and goes up one URL segment - so `../test1` from `/test/123` lands on `/test/test1`

**Reply by avs-doom** [reply]
Thanks for the reply, could you please give me viewing rights?

**Reply by brophdawg11** [reply]
oops - changed it to Public - you should be able to see it now I think

**Reply by avs-doom** [reply]
https://codesandbox.io/p/devbox/cranky-sammet-spfyy3
Thanks, but what's wrong this? A hierarchy was added there and button 2 immediately began to behave differently

**Reply by brophdawg11** [reply]
That's because button two is using `relative:"route"` so it navigates relative to the route hierarchy.  By putting it as a child route you've changed how the relative routing will function.

* You start at the hierarchy `<Route path="test"> -> <Route path=":testId" element={<Test />} />`
* You route to `../test2`
  * `..` means "go up one Route" and takes you up to `<Route path="test">`   
  * `test2` then correctly matches the child `<Route path="test2" element={<div>error</div>} />`

If you want to end up at the root `/test2` component, then you would need to navigate to `../../test2` to go up 2 Route's.

If you are not using any dual-segment paths, such as `path="test/:testId"` then there's really no functional difference betwen `relative:route` and `relative:path` because every Route us one URL segment.
